--- v9/maincpu/kn5000_v9_program.s	2026-03-25 08:56:58.942537061 +0000
+++ v10/maincpu/kn5000_v10_program.s	2026-03-25 08:56:59.632593064 +0000
@@ -315,7 +315,7 @@
 ; ===========================================================================
 SubCPU_Send_Payload:
 	push xiz
-	cpib_da 0xfffeef, 0xff
+	cpib_da ROM_PaddingFF_0x4, 0xff
 	jrl nz, SubCPU_Payload_Done
 	lds32 xiz, 0
 
@@ -344,7 +344,7 @@
 	ld xde, 0x90000
 	call InterCPU_E1_Bulk_Transfer
 	ld xiz, 0x800000
-	cpib_da 0xfffeed, 0xff
+	cpib_da ROM_PaddingFF_0x2, 0xff
 	jr nz, SubCPU_Payload_TransferPart2
 	ld xiz, 0x50000
 	ld xwa, 0x3e0000
@@ -1511,7 +1511,7 @@
 	ldw wa, 0x1e
 	calr DrawQueue_Alloc
 	ld xiz, xhl
-	lda_24 xwa, 0xfb0f31
+	lda_24 xwa, DrawText_PopAndReturn_0x7
 	ld (xhl), xwa
 	ld xwa, (xsp + 16)
 	ld xiy, xwa
@@ -3127,14 +3127,14 @@
 	.set SeqChan_UnhandledCmd_0x03, 0xfd8264
 	.set SeqChan_UnhandledCmd_0x12, 0xfd8273
 	.set AudioInit_PartConfig_Loop_0x26, 0xfe0053
+	.set HdaeRom_DataHandler_0x22, 0xff0270
 	.set HdaeRom_DispatchOffsetTable, CharMap_PermutationPtrTable_B + 540
 	.set HdaeRom_AltDispatchOffsetTable, CharMap_PermutationPtrTable_B + 552
-	.set HdaeRom_DataHandler_0x22, 0xff0270
-	.set _addr24_Free, 0xff0ae4
-	.set _addr24_Malloc, 0xff0e72
-	.set _addr24_Math_MultiplyAccumulate, 0xff0a4e
-	.set _addr24_SendPartDataBlock_Return5, 0xfef9a8
-	.set _addr24_Sprintf_Locked, 0xff0a64
-	.set _addr24_Strcpy, 0xff0f3f
-	.set _addr24_Mem_Copy, 0xff0d8b
+	.set _addr24_Free, 0xff0af2
+	.set _addr24_Malloc, 0xff0e80
+	.set _addr24_Math_MultiplyAccumulate, 0xff0a5c
+	.set _addr24_SendPartDataBlock_Return5, 0xfef9b3
+	.set _addr24_Sprintf_Locked, 0xff0a72
+	.set _addr24_Strcpy, 0xff0f4d
+	.set _addr24_Mem_Copy, 0xff0d99
 	.set NakaData_RomEnd, 0xffffff
